### ISSUE TITLE
Very wrong output

### ISSUE THREAD

**Issue by MekkaSiekka** [mndar/facenet_classifier#8]
Hi, 

I just successfully compiled the code and setup the classification/test data sets, and I am having the following issues:

1. with approx. 60 images, 10 classes, the program only outputs the correct classification IF AND ONLY IF the test set and the training set are exactly the same, including the directory structure. 

2. from my understanding, the grouping and naming of the testing set does not matter, right? 

3. I tried to use 5 groups with 5 images in that to train the classifier, and another 5  different images from the same group(person) for testing, but the result is still terrible. The model I use is downloaded from facenet git repo, and everything compiles w/o warning. 

Any thoughts? 

Could you also post the data sets and models you used for training/testing? 

Best regards,
Chris

**Comment by mndar** [comment]
Which library are you using to extract faces from the images?

**Comment by MekkaSiekka** [comment]
Hi:

I tried both CASIA and LFW, and the frame is 160x160 produced by the mtcnn code provided in the original Facenet github repo. 

Thanks!

**Comment by mndar** [comment]
Which pre-trained model are you using? I'll try with the same model and let you know the results.

**Comment by mndar** [comment]
I tested with the 20180402-114759 and everything seems to be working ok. I have yet to try David Sandberg's MTCNN code. I used Dlib to extract faces from images.
The best results I am getting are with KNN.

**Comment by Dima00782** [comment]
mndar, can you provide your dlib extraction routine? I've asked in other issue and I'm still waiting for answer

**Comment by MekkaSiekka** [comment]
Thanks! I will try that too. That's very helpful.

**Comment by MekkaSiekka** [comment]
**My output with MTCNN extracted face with MLP is as followed:**


/home/xiaojiang/datasets/lfw/lfw_classify/asdf/Tony_Blair_0055.png [6 x 1] 2 Jean_Chretien 0.411344

/home/xiaojiang/datasets/lfw/lfw_classify/asdf/Tony_Blair_0056.png [6 x 1] 1 Laura_Bush 0.428755

/home/xiaojiang/datasets/lfw/lfw_classify/Untitled Folder/Tony_Blair_0055.png [6 x 1] 1 Laura_Bush 0.613469

/home/xiaojiang/datasets/lfw/lfw_classify/Untitled Folder/Tony_Blair_0056.png [6 x 1] 1 Laura_Bush 0.785948

**And the image structure for testing is (40 person in one class for training)**
.
├── Jean_Chretien
│   ├── Jean_Chretien_0001.png
│   ├── Jean_Chretien_0002.png
│   ├── Jean_Chretien_0003.png
│   ├── Jean_Chretien_0004.png
│   ├── Jean_Chretien_0005.png
│   ├── Jean_Chretien_0006.png
│   ├── Jean_Chretien_0007.png
│   ├── Jean_Chretien_0008.png
│   ├── Jean_Chretien_0009.png
│   ├── Jean_Chretien_0010.png
│   ├── Jean_Chretien_0011.png
│   ├── Jean_Chretien_0012.png
│   ├── Jean_Chretien_0013.png
│   ├── Jean_Chretien_0014.png
│   ├── Jean_Chretien_0015.png
│   ├── Jean_Chretien_0016.png
│   ├── Jean_Chretien_0017.png
│   ├── Jean_Chretien_0018.png
│   ├── Jean_Chretien_0019.png
│   ├── Jean_Chretien_0020.png
│   ├── Jean_Chretien_0021.png
│   ├── Jean_Chretien_0022.png
│   ├── Jean_Chretien_0023.png
│   ├── Jean_Chretien_0024.png
│   ├── Jean_Chretien_0025.png
│   ├── Jean_Chretien_0026.png
│   ├── Jean_Chretien_0027.png
│   ├── Jean_Chretien_0028.png
│   ├── Jean_Chretien_0029.png
│   ├── Jean_Chretien_0030.png
│   ├── Jean_Chretien_0031.png
│   ├── Jean_Chretien_0032.png
│   ├── Jean_Chretien_0033.png
│   ├── Jean_Chretien_0034.png
│   ├── Jean_Chretien_0035.png
│   ├── Jean_Chretien_0036.png
│   ├── Jean_Chretien_0037.png
│   ├── Jean_Chretien_0038.png
│   ├── Jean_Chretien_0039.png
│   └── Jean_Chretien_0040.png
├── John_Ashcroft
│   ├── John_Ashcroft_0001.png
│   ├── John_Ashcroft_0002.png
│   ├── John_Ashcroft_0003.png
│   ├── John_Ashcroft_0004.png
│   ├── John_Ashcroft_0005.png
│   ├── John_Ashcroft_0006.png
│   ├── John_Ashcroft_0007.png
│   ├── John_Ashcroft_0008.png
│   ├── John_Ashcroft_0009.png
│   ├── John_Ashcroft_0010.png
│   ├── John_Ashcroft_0011.png
│   ├── John_Ashcroft_0012.png
│   ├── John_Ashcroft_0013.png
│   ├── John_Ashcroft_0014.png
│   ├── John_Ashcroft_0015.png
│   ├── John_Ashcroft_0016.png
│   ├── John_Ashcroft_0017.png
│   ├── John_Ashcroft_0018.png
│   ├── John_Ashcroft_0019.png
│   ├── John_Ashcroft_0020.png
│   ├── John_Ashcroft_0021.png
│   ├── John_Ashcroft_0022.png
│   ├── John_Ashcroft_0023.png
│   ├── John_Ashcroft_0024.png
│   ├── John_Ashcroft_0025.png
│   ├── John_Ashcroft_0026.png
│   ├── John_Ashcroft_0027.png
│   ├── John_Ashcroft_0028.png
│   ├── John_Ashcroft_0029.png
│   ├── John_Ashcroft_0030.png
│   ├── John_Ashcroft_0031.png
│   ├── John_Ashcroft_0032.png
│   ├── John_Ashcroft_0033.png
│   ├── John_Ashcroft_0034.png
│   ├── John_Ashcroft_0035.png
│   ├── John_Ashcroft_0036.png
│   ├── John_Ashcroft_0037.png
│   ├── John_Ashcroft_0038.png
│   ├── John_Ashcroft_0039.png
│   └── John_Ashcroft_0040.png
├── Laura_Bush
│   ├── Laura_Bush_0002.png
│   ├── Laura_Bush_0004.png
│   ├── Laura_Bush_0005.png
│   ├── Laura_Bush_0006.png
│   ├── Laura_Bush_0007.png
│   ├── Laura_Bush_0008.png
│   ├── Laura_Bush_0009.png
│   ├── Laura_Bush_0010.png
.....with more classes
`

**The testing classes are simply**
.
├── asdf
│   ├── Tony_Blair_0055.png
│   └── Tony_Blair_0056.png
└── Untitled Folder
    ├── Tony_Blair_0055.png
    └── Tony_Blair_0056.png

So the problem I have now is that the MLP algorithm simply classifies the image to the wrong person; do you have any clue? Also, as you are saying that you are using the dlib library to extract face, is there a specific way to do it?
